js/src/picc.js

```diff
@@ -0,0 +1,1621 @@
+   * });
+   *
+   * @param {Object}    urls      a map of properties to either URIs or
+   *                              Arrays, in which the first element is the
+   *                              function to call and the rest are arguments.
+   * @param {Function}  callback  the callback function:
+   *                              `callback(error, data)`
+   */
+  API.getAll = function(urls, done) {
+    Object.keys(urls).forEach(function(key) {
+      var url = urls[key];
+      urls[key] = Array.isArray(url)
+        ? function(done) {
+          var method = url.shift();
+          if (typeof method === 'string') {
+            method = API[method];
+          }
+          url.push(done);
+          return method.apply(API, url);
+        }
+        : function(done) {
+          return API.get(url, done);
+        };
+    });
+    // console.log('getAll:', urls);
+    return async.parallel(urls, done);
+  };
+
+  /**
+   * add the API key (if set) to either an object or string of query
+   * parameters, and return the parameters as a query string.
+   *
+   * @param {String|Object} params
+   * @return {String}
+   */
+  function addAPIKey(params) {
+    var param = 'api_key';
+    if (typeof params === 'object') {
+      if (API.key) params[param] = API.key;
+      // collapse arrays into comma-separated strings
+      // per the API
+      collapseArrays(params);
+      params = querystring.stringify(params);
+    } else if (API.key) {
+      params += ['&', param, '=', API.key].join('');
+    }
+    return params;
+  }
+
+  /**
+   * Join an array of strings with a `glue` string and de-dupe repeated glue
+   * strings.
+   *
+   * @param {Array} list
+   * @param {String} glue
+   * @return {String}
+   */
+  function join(list, glue) {
+    for (var i = 0; i < list.length; i++) {
+      var str = String(list[i]);
+      if (str.charAt(0) === glue) {
+        list[i] = str.substr(1);
+      } else if (str.charAt(str.length - 1) === glue) {
+        list[i] = str.substr(0, str.length - 1);
+      }
+    }
+    return list.join(glue);
+  }
+
+  /**
+   * Iterate over the keys of an `object` and replace any that have Array
+   * values with strings joined with `glue`.
+   *
+   * @param {Object} obj
+   * @param {String?} glue
+   * @return {Object} the original object
+   */
+  function collapseArrays(obj, glue) {
+    if (!glue) glue = ',';
+    for (var key in obj) {
+      if (Array.isArray(obj[key])) {
+        obj[key] = obj[key].join(glue);
+      }
+    }
+    return obj;
+  }
+
+  return API;
+})();
+
+// this is our "n/a" string that we display for null numeric values
+var NA = '--';
+
+/**
+ * This is our format generator. Its methods are format generators for
+ * specific types of values, and they take a key in the data object to
+ * format. For instance:
+ *
+ * @example
+ * var formatFoo = format.percent('foo');
+ * formatFoo({foo: 0.5}) -> '50%'
+ */
+picc.format = (function() {
+  var formatter = function(fmt, _empty) {
```

Comment:
And I may be missing something, but why define as an IIFE instead of a plain old CommonJS module?